### PR TITLE
refactor address

### PR DESCRIPTION
**Pull request by masterjedy**
Signed-off-by: Yura Zarudniy <zarudniy@soramitsu.co.jp>

### Description of the Change
Address refactored.

### Benefits
Address refactored.

### Possible Drawbacks 
None.

### CODE REVIEW

File: core/primitives/address/address_builder.hpp [hunk 1]
```diff
@@ -6,11 +6,11 @@
 #ifndef CPP_FILECOIN_CORE_PRIMITIVES_ADDRESS_ADDRESS_BUILDER_HPP
 #define CPP_FILECOIN_CORE_PRIMITIVES_ADDRESS_ADDRESS_BUILDER_HPP
 
-#include <libp2p/crypto/secp256k1_types.hpp>
+#include "crypto/secp256k1_provider/secp256k1_provider.hpp"
 
 #include "common/outcome.hpp"
 #include "crypto/bls_provider/bls_types.hpp"
-#include "primitives/address.hpp"
+#include "address.hpp"
```
Comment (igor-egorov) [inline core/primitives/address/address_builder.hpp]: I bet it is not clang-formatted (the order of includes)

File: core/primitives/address/impl/address_builder_impl.cpp [hunk 2]
```diff
@@ -6,28 +6,31 @@
...
   using crypto::blake2b::blake2b_160;
+  using primitives::address::decode;
+  using primitives::address::encode;
+  using primitives::address::Protocol;
 
   outcome::result<Address> AddressBuilderImpl::makeFromSecp256k1PublicKey(
       Network network, const Sec256k1PublicKey &public_key) noexcept {
     OUTCOME_TRY(hash, blake2b_160(public_key));
-    std::vector<uint8_t> sec256k1_bytes{network, primitives::SECP256K1};
+    std::vector<uint8_t> sec256k1_bytes{network, Protocol::SECP256K1};
```
Comment (igor-egorov) [inline core/primitives/address/impl/address_builder_impl.cpp]: not an issue, just to know. why Secp256k1 is all capitalized?

Comment (igor-egorov) [reply core/primitives/address/impl/address_builder_impl.cpp]: BLS is ok since it is an acronym

**Comment by Alexey-N-Chernyshov** [comment]
Also consider renaming `core/crypto/bls_provider' to 'core/crypto/bls'.

File: core/primitives/address/impl/address_builder_impl.cpp (hunk 2 above)
Comment (masterjedy) [reply core/primitives/address/impl/address_builder_impl.cpp]: it is also an acronym) anyway, if memory serves me right, we decided to use upper-case in constants. sometimes it applies to enums members, but it is not fully legal.

File: core/primitives/address/address_builder.hpp (hunk 1 above)
Comment (masterjedy) [reply core/primitives/address/address_builder.hpp]: fixed

**Pull request merged**